contributed/MaxPool2D.py

```diff
@@ -0,0 +1,134 @@
+"""
+MaxPool2d NKI with benchmark
+
+Pretty similar to https://github.com/aws-neuron/nki-samples/blob/main/src/nki_samples/tutorials/average_pool2d/average_pool2d_nki_kernels.py
+
+
+"""
+
+import torch
+import torch.nn.functional as F
+import numpy as np
+import time
+import neuronxcc.nki as nki
+import neuronxcc.nki.language as nl
+from neuronxcc.nki.typing import tensor
+
+@nki.jit
+def maxpool_kernel(in_tensor, kernel_size, padding=0, stride=None):
+    """NKI kernel to compute a 2D max-pool operation
+    
+    Args:
+        in_tensor: Input tensor of shape (C, H, W)
+        kernel_size: Size of pooling window
+        padding: Integer padding size.  Defaults to zero
+        stride: Stride of pooling window. If None, defaults to pool_size
+        pad should be at most half of effective kernel size, 
+        Only handles integer pad at this point
+
+        Todo:  add an assert for the pad/kernel size like pytorch does
+            Also, consider a (1,2) type format for padding in addition to an integer.  
+            Maybe the same thing for kernel size?
+            Add a batch dimension
+    """
+    if stride is None:
+        stride = kernel_size
+        
+    # Get input/output dimensions
+    sz_cin, sz_hin, sz_win = in_tensor.shape
+    sz_hout = (sz_hin + 2*padding - kernel_size) // stride + 1
+    sz_wout = (sz_win + 2*padding - kernel_size) // stride + 1
+    
+        
+    # Create output tensor
+    out_tensor = nl.ndarray((sz_cin, sz_hout, sz_wout), dtype=in_tensor.dtype,
+                           buffer=nl.shared_hbm)
+
+    # Set relevant sizes
+    sz_p = sz_cin
+
+    # Generate pool index patterns with stride
+    i0 = nl.arange(sz_p)[:, None, None, None, None]  # Channel dim
+    i1 = nl.arange(sz_hout)[None, :, None, None, None]  # Output height
+    i2 = nl.arange(sz_wout)[None, None, :, None, None]  # Output width
+    i3 = nl.arange(kernel_size)[None, None, None, :, None]  # Pool height
+    i4 = nl.arange(kernel_size)[None, None, None, None, :]  # Pool width
+
+    # Load input data
+    in_tile: tensor[sz_p, sz_hin, sz_win] = nl.load(in_tensor)
```

Comment:
per the docs here, your partition dimension must be the first dimension. These should be 2d tiles. We're deprecating block dimension on SBUF.

https://awsdocs-neuron.readthedocs-hosted.com/en/latest/general/nki/api/generated/nki.language.load.html#nki.language.load